### ISSUE TITLE
Installing plugins in logstash 1.5.0 using proxy ?

### ISSUE THREAD

**Issue by JohanGregoire** [elastic/logstash#3314]
Hello,

I'm trying to install a custom plugin into my logstash 1.5.0.
According to the documentation i should be able to do so with this command: bin/plugin install logstash-filter-cidr

However when I try it I get this:

root@elky:/usr/local/logstash-1.5.0# bin/plugin install logstash-filter-cidr

/usrlocal/logstash-1.5.0/vendor/jruby/lib/ruby/shared/rubygems/specification.rb:102 warning: Unrecognized time zone: localtime
io/console not supported; tty will not be manipulated
Validating logstash-filter-cidr
Plugin logstash-filter-cidr does not exist
ERROR: Installation aborted, verification failed for logstash-filter-cidr

We are behind a proxy and I set this environment to get through it:
http_proxy=http://www:8080

Can anyone help me or point me in the right direction so I can get this plugin installed?

Thanks

Regards,
Johan Gregoire

**Comment by purbon** [comment]
I'm investigating this issue,  but I would need more details of your proxy configuration to be able to reproduce it. Is there any particularity of the proxy you're going throw? can you give me more details so I can for example create an squid/varnish proxy and reproduce it this?

Thanks a lot

**Comment by ph** [comment]
@JohanGregoire I did a small check locally, I was able to make them run successfully:

```
#From logstash
~/e/s/logstash-1.5.0 ❯❯❯ export http_proxy=http://127.0.0.1:3128
~/e/s/logstash-1.5.0 ❯❯❯ bin/plugin install logstash-filter-cidr                                                                                                                                                                                                                                                                                                        
Validating logstash-filter-cidr
Installing logstash-filter-cidr
Installation successful

```
```
# My squid access log.
1432655406.264   8299 127.0.0.1 TCP_MISS/200 4689 CONNECT s3.amazonaws.com:443 - HIER_DIRECT/54.231.10.8 -
1432655411.445    338 127.0.0.1 TCP_MISS/200 13422 CONNECT rubygems.global.ssl.fastly.net:443 - HIER_DIRECT/199.27.76.249 -
1432655411.445  12324 127.0.0.1 TCP_MISS/200 1882674 CONNECT bundler.rubygems.org:443 - HIER_DIRECT/107.20.214.249 -
1432655411.445  14632 127.0.0.1 TCP_MISS/200 5409 CONNECT api.rubygems.org:443 - HIER_DIRECT/54.186.104.15 -
1432655411.445    968 127.0.0.1 TCP_MISS/200 5425 CONNECT rubygems.org:443 - HIER_DIRECT/54.186.104.15 -
```

**Comment by ph** [comment]
@JohanGregoire Maybe the Validation call doesnt go through the proxy?

Can you check if this command work?
```
bin/plugin install --no-verify logstash-filter-cidr
```

**Comment by ph** [comment]
The last command bypass the validation, the validation check if this is an official logstash plugins.

**Comment by JohanGregoire** [comment]
Hello,

Thanks for the quick replies. I've been testing some more.
When I use the no-verifiy option I get another error:
```
export http_proxy=http://www:8080

bin/plugin install --no-verify logstash-filter-cidr
```
Output:
```
/usrlocal/logstash-1.5.0/vendor/jruby/lib/ruby/shared/rubygems/specification.rb:102 warning: Unrecognized time zone: localtime
io/console not supported; tty will not be manipulated
Installing logstash-filter-cidr
Error Bundler::HTTPError, retrying 1/10
Could not fetch specs from https://rubygems.org/
Error Bundler::HTTPError, retrying 2/10
Could not fetch specs from https://rubygems.org/
Error Bundler::HTTPError, retrying 3/10
```

*When I use this:*
(only difference is the port-number)
```
export http_proxy=http://www:3128

bin/plugin install --no-verify logstash-filter-cidr

```
I get this:
```
/usrlocal/logstash-1.5.0/vendor/jruby/lib/ruby/shared/rubygems/specification.rb:102 warning: Unrecognized time zone: localtime
io/console not supported; tty will not be manipulated
Installing logstash-filter-cidr
Error Bundler::InstallError, retrying 1/10
An error occurred while installing logstash-filter-cidr (0.1.6), and Bundler cannot continue.
Make sure that `gem install logstash-filter-cidr -v '0.1.6'` succeeds before bundling.
Error Bundler::InstallError, retrying 2/10
An error occurred while installing logstash-filter-cidr (0.1.6), and Bundler cannot continue.
Make sure that `gem install logstash-filter-cidr -v '0.1.6'` succeeds before bundling.
root@elky:/usr/local/logstash-1.5.0#

```

So I might still be having proxy issues here.
Or do you guys think of something else?

Regards,
Johan Gregoire

**Comment by purbon** [comment]
@JohanGregoire can you share a bit more your proxy peculiarities ? how is your peculiar config? is your www machine reached properly ? 

I guess you're working with linux, right?

**Comment by JohanGregoire** [comment]
Hello,

Is there a way to manually download a plugin and install it?

I'm working on Solaris 11.
I'm going to check with out network-admins what is being blocked by the proxy.

Regards,
Johan Gregoire

**Comment by purbon** [comment]
You can actually go to rubygems [1] and dowload, check for the download link. You can do that for any plugin.

On the other side, keep in mind we don't really test and support Solaris, there is no really way for us to test in there. Although please, let us know if your proxy has something block, or is molesting your connection to rubygems somehow.

/cheers

[1] https://rubygems.org/gems/logstash-filter-cidr

**Comment by JohanGregoire** [comment]
Hello,

For now I went with the manual install and it went flawlessly :)

Thanks for the help!
When I get some more time, I'll revisit the proxy issue.

Regards,
Johan Gregoire

**Comment by purbon** [comment]
@JohanGregoire as you problem is workaround and you're using solaris (a platform where we don't really do tests), I'm open to close this issue until you can grab mode intel on your proxy config, makes sense to you?

PD: I still find very interesting to know more about your proxy issues, so please don't forget to share it if possible ;-)

**Comment by purbon** [comment]
@JohanGregoire closing this issue for now, as soon as you get more intel on your proxy config, feel free to create a new issue with it, or also reopen/update this one. Thanks a lot for your feedback.

**Comment by smougenot** [comment]
It might help others: you may need to proxy both HTTP and HTTPS

    export http_proxy=http://user:password@host:port
    export https_proxy=http://user:password@host:port
    /opt/logstash/bin/plugin install logstash-???-??

**Comment by agacem** [comment]
I confirme
This work's !!
Thx.
export http_proxy=http://user:password@host:port
export https_proxy=http://user:password@host:port

**Comment by JuanCarniglia** [comment]
A little advice (fought against this one). When you run a command using "sudo", the environment variables that you though were set while being a "regular user", are not necessarily set for the new "super user" session. So, if you encounter this kind of problems where a variable is supposed to be set, but is ignored, check that these variables are actually set.

You can either go "su -" and then set your variables (now being super user) or you can try executing "sudo -E {command}" which according to the man page, should preserve user's vaiables.
